src/components/Table/RowState.ts

```diff
@@ -161,6 +177,30 @@ export class RowState {
       localStorage.setItem(this.persistCollapse, JSON.stringify(collapsedIds));
     }
   }
+
+  private getVisibleImmediateChildrenSelectedStates(
+    children: GridDataRow<any>[],
+    map: Map<string, SelectedState>,
+  ): SelectedState[] {
+    return children
+      .filter((row) => row.id !== "header" && this.visibleRows.has(row.id))
+      .map((row) => map.get(row.id) || this.getSelected(row.id));
+  }
+
+  // Recursively traverse through rows to determine selected state of parent rows based on children
+  private setNestedSelectedStates(row: GridDataRow<any>, map: Map<string, SelectedState>): SelectedState[] {
```

Comment:
I thought about trying to consolidate this function and the verbosely named `getVisibleImmediateChildrenSelectedStates` function, but there are times when we don't need to recursively look through children, so I thought its better to keep them separate and only use this recursive function when needed.

Reply:
Yeah, I'm fine keeping two separate functions.

Maybe add an in-source comment that "we return a `SelectedState[]` so that hidden rows don't "have a vote" about the full/partial checked-ness of their parent.

An alternative to returning `SelectedState[]` would be `SelectedState | undefined`, but then we'd have to filter out `undefined` in `derivedParentSelected`.

Or add like an addition `SelectedState` value of `"hidden"` and again teach `derivedParentSelected` to ignore `hidden` rows... like:

```
const allVisibleChecked = children.every(child => child === "hidden" || child === "unchecked');
const allVisibleUnchecked = children.every(child => child === "hidden" || child === "unchecked");
```

...I kind of like the explicitness of adding "hidden", but the current `SelectedState[]` is fine too.